src/wp-includes/functions.php

```diff
@@ -8860,6 +8860,40 @@ function wp_fuzzy_number_match( $expected, $actual, $precision = 1 ) {
 	return abs( (float) $expected - (float) $actual ) <= $precision;
 }
 
+
+/**
+ * Replaces trailing period, if any, with encoded entity in URLs.
+ *
+ * If the URL ends with a period, replace it with %2E to avoid
+ * misinterpretation from some email clients.
+ *
+ * @since 6.6.0
+ * @access private
+ *
+ * @param string $url The URL to encode.
+ * @return string The new encoded URL.
+ */
+function _encode_query_arg_ending_period( $url ) {
+	if ( ! is_string( $url ) ) {
+		return '';
+	}
+
+	$url = trim( $url );
+	if ( '' === $url ) {
+		return '';
+	}
+
+	$url = rawurlencode( $url );
```

Comment:
note that we don't want to call `rawurlencode()` on a full URL. it's only meant to encode the parts after the domain, and we want to avoid double-encoding it, since if we call this on an already-escaped URL, we'll end up escaping the escapes and corrupting it.

Reply:
Fair enough, though I am going to update my pull request to match your other very elegant solution